### PR TITLE
[#166791775] Added tabs in services section

### PR DESCRIPTION
**Pull request by laurapugliese**
Created new screen for home services visible only if experimental features are enabled

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#166791775](https://www.pivotaltracker.com/story/show/166791775): Come CIT voglio vedere la sezione servizi divisa in tre tab

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by laurapugliese** [comment]
In this PR a new screen has been created for the new services home. The new screen is visible **ONLY if the experimental features are enabled**:

tabs

**Comment by Undermaken** [comment]
@laurapugliese I run this branch on my device and I have a doubt: 
Schermata 2019-06-21 alle 14 59 19
When I am in this screen I can't go back in preferences. Is it wanted?

**Comment by laurapugliese** [comment]
> @laurapugliese I run this branch on my device and I have a doubt:
> 
> Schermata 2019-06-21 alle 14 59 19
> When I am in this screen I can't go back in preferences. Is it wanted?

@Undermaken  you are right. I have not inserted the back because I am waiting to coordinate with @lmorelli986 and @fpersico because there is the PR https://github.com/teamdigitale/io-app/pull/1151 that integrates the services in the tabBar (in this case the back does not it is necessary). So based on the order in which these PRs will have to be changed (we also have other conflicts on the names of the routes). Perhaps it is more correct that I integrate the back for the moment and then in case we remove it later. What do you think about it?

**Comment by Undermaken** [comment]
> > @laurapugliese I run this branch on my device and I have a doubt:
> > Schermata 2019-06-21 alle 14 59 19
> > When I am in this screen I can't go back in preferences. Is it wanted?
> 
> @Undermaken you are right. I have not inserted the back because I am waiting to coordinate with @lmorelli986 and @fpersico because there is the PR #1151 that integrates the services in the tabBar (in this case the back does not it is necessary). So based on the order in which these PRs will have to be changed (we also have other conflicts on the names of the routes). Perhaps it is more correct that I integrate the back for the moment and then in case we remove it later. What do you think about it?

Ok I get it!
Anyway if this PR should be released as experimental feature I guess a back is needed.
Does it have sense?

**Comment by laurapugliese** [comment]
> > > @laurapugliese I run this branch on my device and I have a doubt:
> > > Schermata 2019-06-21 alle 14 59 19
> > > When I am in this screen I can't go back in preferences. Is it wanted?
> > 
> > 
> > @Undermaken you are right. I have not inserted the back because I am waiting to coordinate with @lmorelli986 and @fpersico because there is the PR #1151 that integrates the services in the tabBar (in this case the back does not it is necessary). So based on the order in which these PRs will have to be changed (we also have other conflicts on the names of the routes). Perhaps it is more correct that I integrate the back for the moment and then in case we remove it later. What do you think about it?
> 
> Ok I get it!
> Anyway if this PR should be released as experimental feature I guess a back is needed.
> Does it have sense?

@Undermaken Ok I insert the back for the moment. Then eventually I remove it later. ;)

File: ts/screens/services/ServicesHomeScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  private animatedScrollPositions: ReadonlyArray<Animated.Value> = [
+    new Animated.Value(0),
+    new Animated.Value(0),
+    new Animated.Value(0)
+  ];
+
+  // tslint:disable-next-line: readonly-array
+  private scollPositions: number[] = [0, 0, 0];
```
Comment (Undermaken) [inline ts/screens/services/ServicesHomeScreen.tsx]:
```suggestion
  private scrollPositions: number[] = [0, 0, 0];
```
I also checked **MessagesHomeScreen** and I noticed this class member is never used. Consider to remove it if it won't be useful for next developments

**Comment by laurapugliese** [comment]
@Undermaken as suggested I integrated the back. Below the final result:

servizi

File: ts/screens/services/ServicesHomeScreen.tsx (hunk 1 above)
Comment (laurapugliese) [reply ts/screens/services/ServicesHomeScreen.tsx]: @Undermaken Yes it is not necessary at the moment. I can remove it and add it when the list of services is integrated

**Comment by fpersico** [comment]
@laurapugliese Services is now a totally different tab we need to change this PR to use the new layout.

**Comment by laurapugliese** [comment]
After integration of new home services in tab:

tab_lista_servizi_compressed

**Comment by laurapugliese** [comment]
> @laurapugliese Services is now a totally different tab we need to change this PR to use the new layout.

@fpersico Done.

File: ts/screens/preferences/ServicesScreen.tsx [hunk 2]
```diff
@@ -163,10 +185,11 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
     dispatch(contentServiceLoad.request(serviceId)),
   navigateToServiceDetailsScreen: (
     params: InferNavigationParams<typeof ServiceDetailsScreen>
-  ) => dispatch(navigateToServiceDetailsScreen(params))
+  ) => dispatch(navigateToServiceDetailsScreen(params)),
+  navigateToServiceHomeScreen: () => dispatch(navigateToServicesHomeScreen())
```
Comment (fpersico) [inline ts/screens/preferences/ServicesScreen.tsx]:
```suggestion
  navigateToServicesHomeScreen: () => dispatch(navigateToServicesHomeScreen())
```

File: ts/screens/preferences/ServicesScreen.tsx [hunk 3]
```diff
@@ -50,6 +53,22 @@ class ServicesHomeScreen extends React.Component<Props> {
     });
   };
 
+  public componentWillMount() {
+    this.checkForExperimentalFeatures();
+  }
+
+  private handleWillFocus = () => {
+    this.checkForExperimentalFeatures();
```
Comment (fpersico) [inline ts/screens/preferences/ServicesScreen.tsx]: You are creating a loop here.
1) You come here with experimentalFeatures flag enabled.
2) You will be redirected to the NEW ServiceHome
3) Than an Android you click the hardware back you will return to the old ServiceHome (you are not removing the screen from the history) but this method will bring you again to the new ServiceHome.

**Pull request merged**